modules/tinymce/src/core/main/ts/content/TransparentElements.ts

```diff
@@ -20,16 +21,97 @@ const updateTransparent = (blocksSelector: string, transparent: Element) => {
     if (transparent.getAttribute('data-mce-selected') === 'inline-boundary') {
       transparent.removeAttribute('data-mce-selected');
     }
+
+    return true;
   } else {
     transparent.removeAttribute(transparentBlockAttr);
+    return false;
   }
 };
 
-export const updateChildren = (schema: Schema, scope: Element): void => {
+const updateBlockStateOnChildren = (schema: Schema, scope: Element): Element[] => {
   const transparentSelector = makeSelectorFromSchemaMap(schema.getTransparentElements());
   const blocksSelector = makeSelectorFromSchemaMap(schema.getBlockElements());
 
-  Arr.each(scope.querySelectorAll(transparentSelector), (transparent) => updateTransparent(blocksSelector, transparent));
+  return Arr.filter(scope.querySelectorAll(transparentSelector), (transparent) => updateTransparent(blocksSelector, transparent));
+};
+
+const trimEdge = (el: DocumentFragment, leftSide: boolean) => {
+  const childName = leftSide ? 'lastChild' : 'firstChild';
```

Comment:
Could we pick a different name for this variable? `childname` makes it seem like we're looking for specific, named nodes instead of if we're going to the left or right?

Reply:
Good point, maybe `childPropertyName`.